### PR TITLE
Added go pipeline for binaries

### PR DESCRIPTION
**Pull request by nkierpiec**
- Uses goreleaser to build and artifactory to publis
- Uses github to release binaries by OS type as well as singular

Signed-off-by: Nathaniel Kierpiec <nkierpiec@chef.io>

## Description
This will help create a streamlined process for how we build and release our go binaries

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [x ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .expeditor/create_release.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "go-binaries-local/*"
+  "go-binaries/${os}/${arch}/"
+
+  for file in go-binaries/${os}/${arch}/*
+  do
+    if [[ ${os} == "windows" ]]; then
+      IFS='.' read -r -a parts <<< "${file}"
+      util_name=${parts[0]}
+
+      zip -r "ci_stuido_common_${os}_${arch}.zip" go-binaries/${os}/${arch}/
```
Comment (tduffield) [inline .expeditor/create_release.sh:28]: sp: studio vs stuido

**Review by tduffield — APPROVED** [review]
Fix the typos and we should be good :)

**Review by jeremiahsnapp — APPROVED** [review]
Fantastic work @nkierpiec !

**Review by jacobboykin — APPROVED** [review]

**Review by jacobboykin — APPROVED** [review]

File: internal/pkg/secrets/client_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"net"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/vault"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetSecret(t *testing.T) {
```
Comment (jeremiahsnapp) [inline internal/pkg/secrets/client_test.go:14]: Tests, Yay!!! 🎉

**Review by jeremiahsnapp — APPROVED** [review]

**Pull request merged**